### PR TITLE
Tgp 1443/implement navigation for change record warning page goods description and page commodity code

### PR DESCRIPTION
**Pull request by pauloncast**
Covers the following:

- [GP-1443](https://jira.tools.tax.service.gov.uk/browse/TGP-1443) - Implement Navigation for change record warning page-commodity code
- [TGP-1474](https://jira.tools.tax.service.gov.uk/browse/TGP-1474) - Implement Navigation for change record warning page-Goods description

### CODE REVIEW

File: app/controllers/SingleRecordController.scala [hunk 1]
```diff
@@ -42,10 +42,10 @@ class SingleRecordController @Inject() (
     goodsRecordConnector.getRecord(request.eori, recordId).map { record =>
       val detailsList = SummaryListViewModel(
         rows = Seq(
-          TraderReferenceSummary.row(record.traderRef, recordId, NormalMode),
-          GoodsDescriptionSummary.row(record.goodsDescription, recordId, NormalMode),
-          CountryOfOriginSummary.row(record.countryOfOrigin, recordId, NormalMode),
-          CommodityCodeSummary.row(record.commodityCode, recordId, NormalMode),
+          TraderReferenceSummary.row(record.traderRef, recordId, CheckMode),
```
Comment (besscerule) [inline app/controllers/SingleRecordController.scala]: should be normal mode as it is the start of a journey - not a check your answers page

File: app/navigation/Navigator.scala [hunk 2]
```diff
@@ -197,6 +197,8 @@ class Navigator @Inject() () {
     case p: EmailPage                              => _ => routes.CyaRequestAdviceController.onPageLoad(p.recordId)
     case p: AssessmentPage                         => navigateFromAssessmentCheck(p)
     case p: HasSupplementaryUnitPage               => navigateFromHasSupplementaryUnitCheck(p.recordId)
+    case p: HasCommodityCodeChangePage             => navigateFromHasCommodityCodeChangeCheck(p.recordId)
```
Comment (besscerule) [inline app/navigation/Navigator.scala]: should be in the normal section, not check section

File: app/navigation/Navigator.scala [hunk 3]
```diff
@@ -347,6 +349,26 @@ class Navigator @Inject() () {
       }
       .getOrElse(routes.JourneyRecoveryController.onPageLoad())
 
+  private def navigateFromHasCommodityCodeChangeCheck(recordId: String)(answers: UserAnswers): Call =
+    answers
+      .get(HasCommodityCodeChangePage(recordId))
+      .map {
+        case true  =>
+          routes.CommodityCodeController.onPageLoadUpdate(CheckMode, recordId)
```
Comment (besscerule) [inline app/navigation/Navigator.scala]: needs to be normal mode

File: app/navigation/Navigator.scala [hunk 4]
```diff
@@ -347,6 +349,26 @@ class Navigator @Inject() () {
...
+        case false => routes.SingleRecordController.onPageLoad(recordId)
+      }
+      .getOrElse(routes.JourneyRecoveryController.onPageLoad())
+
+  private def navigateFromHasGoodDescriptionChangeCheck(recordId: String)(answers: UserAnswers): Call =
+    answers
+      .get(HasGoodDescriptionChangePage(recordId))
+      .map {
+        case true  =>
+          routes.GoodsDescriptionController.onPageLoadUpdate(CheckMode, recordId)
```
Comment (besscerule) [inline app/navigation/Navigator.scala]: needs to be normalmode

File: app/viewmodels/checkAnswers/CommodityCodeSummary.scala [hunk 5]
```diff
@@ -45,7 +45,7 @@ object CommodityCodeSummary {
       key = "commodityCode.checkYourAnswersLabel",
       value = ValueViewModel(HtmlFormat.escape(value).toString),
       actions = Seq(
-        ActionItemViewModel("site.change", routes.CommodityCodeController.onPageLoadUpdate(mode, recordId).url)
+        ActionItemViewModel("site.change", routes.HasCommodityCodeChangeController.onPageLoad(mode, recordId).url)
```
Comment (besscerule) [inline app/viewmodels/checkAnswers/CommodityCodeSummary.scala:48]: this produces strange behaviour when the user clicks the change link at the end of this journey before they save their new answer - it goes back to the warning page - i think it should go back to the commodity code page - but double check with amaka

File: app/viewmodels/checkAnswers/GoodsDescriptionSummary.scala [hunk 6]
```diff
@@ -45,7 +45,7 @@ object GoodsDescriptionSummary {
       key = "goodsDescription.checkYourAnswersLabel",
       value = ValueViewModel(HtmlFormat.escape(value).toString),
       actions = Seq(
-        ActionItemViewModel("site.change", routes.GoodsDescriptionController.onSubmitUpdate(mode, recordId).url)
+        ActionItemViewModel("site.change", routes.HasGoodDescriptionChangeController.onPageLoad(mode, recordId).url)
```
Comment (besscerule) [inline app/viewmodels/checkAnswers/GoodsDescriptionSummary.scala]: same issue as above

**Review by besscerule — CHANGES_REQUESTED** [review]
Also once the data is saved it should navigate to goods record page currently it goes to the homepage

File: app/controllers/SingleRecordController.scala (hunk 1 above)
Comment (pauloncast) [reply app/controllers/SingleRecordController.scala]: Updated

File: app/navigation/Navigator.scala (hunk 2 above)
Comment (pauloncast) [reply app/navigation/Navigator.scala]: Updated

File: app/navigation/Navigator.scala (hunk 3 above)
Comment (pauloncast) [reply app/navigation/Navigator.scala]: Updated

File: app/viewmodels/checkAnswers/CommodityCodeSummary.scala (hunk 5 above)
Comment (pauloncast) [reply app/viewmodels/checkAnswers/CommodityCodeSummary.scala:48]: The behaviour should be alright now

File: app/viewmodels/checkAnswers/GoodsDescriptionSummary.scala (hunk 6 above)
Comment (pauloncast) [reply app/viewmodels/checkAnswers/GoodsDescriptionSummary.scala]: Updated

File: app/navigation/Navigator.scala (hunk 4 above)
Comment (pauloncast) [reply app/navigation/Navigator.scala]: updated

**Pull request closed without merging**